addon/components/base-layer.js

```diff
@@ -156,6 +156,28 @@ export default Ember.Component.extend(
     },
 
     /**
+      Handles 'flexberry-map:query' event of leaflet map.
+
+      @method _query
+      @param {Object} e Event object.
+      @param {Object} layerLinks Array contains layer links model, use for filter searched layers
+      @param {Object} queryFilter Object with query filter paramteres
+      @param {Object[]} results.features Array containing leaflet layers objects
+      or a promise returning such array.
+    */
+    _query(e) {
+      let layerId = this.get('layer.id').toString();
+      let layerLinks = e.layerLinks.filter(link => link.get('layer.id').toString() === layerId);
```

Comment:
If we need layer links related only to current layer, why not to filter them in map-route where layer's hierarchy is loaded, & place them into each layer (related to them) as unstored property (like 'layers' for child layers in each layer model)?